src/process_utility.c

```diff
@@ -721,14 +801,20 @@ process_grant_and_revoke_role(ProcessUtilityArgs *args)
 }
 
 static void
-process_drop(Node *parsetree)
+process_drop(ProcessUtilityArgs *args)
 {
-	DropStmt *stmt = (DropStmt *) parsetree;
+	DropStmt *stmt = (DropStmt *) args->parsetree;
 
 	switch (stmt->removeType)
 	{
 		case OBJECT_TABLE:
-			process_drop_hypertable_chunks(stmt);
+			process_drop_hypertable_chunks(args, stmt);
+			break;
+		case OBJECT_INDEX:
```

Comment:
I think this actually has to go through `process_ddl_sql_drop`, and maybe use another hook. The current approach would miss all CASCADING drops.  I think that merging this in the current state is dangerous since it gives the impression that things drops are caught when they aren't.